### ISSUE TITLE
[3.0.3] [console] Notice on QuestionHelper

### ISSUE THREAD

**Issue by shopblocks** [symfony/symfony#18001]
If I define my code as follows:

```php
$question = new ChoiceQuestion('Is this server for Production or Staging', [
    1 => 'production',
    2 => 'staging',
], '1');

$choice = $this->question->ask($input, $output, $question);
```

If i run this question and then press the up arrow, I get undefined offset '0' error. 

I want to be able to number these options myself. If i dont number, then everything is fine, but if i do number then this is the result we get.

Full error: `PHP Notice:  Undefined offset: 0 in /.../console/vendor/symfony/console/Helper/QuestionHelper.php on line 302`

**Comment by hhamon** [comment]
I confirm this notice error message when trying your code on Symfony 3.0.3.

**Comment by shopblocks** [comment]
I see the merge references 2.7, does  this automatically get ported to 3.0 too?

**Comment by jakzal** [comment]
@shopblocks not automatically, but we merge older branches to the newer ones regularly, so the fix will eventually get to 3.0.
